src/Compilers/CSharp/Test/Symbol/Symbols/CustomModifiersTests.cs

```diff
@@ -1756,7 +1756,7 @@ interface ITest4<T, U>
             compilation.VerifyDiagnostics();
         }
 
-        [ConditionalFact(typeof(ClrOnly), typeof(DesktopOnly))]
+        [ConditionalFact(typeof(ClrOnly), typeof(DesktopOnly), Skip = "An unexpected use of nullability is found in source module")]
```

Comment:
@cston `VerifyUsesOfNullability` finds a use of nullability in this test: "void CL3.Test(System.Collections.Generic.Dictionary<dynamic!, dynamic!>, System.Collections.Generic.Dictionary<dynamic!, dynamic!>, System.Collections.Generic.Dictionary<dynamic!, dynamic!>)"

This got me wondering. Why are we expecting no symbols with nullability when the feature is disabled? I would expect binding to behave the same regardless of feature flags (I'd expect only diagnostics should be different).

Reply:
If a Compilation where the feature is disabled is used as a CompilationReference by another Compilation where the feature is enabled, the Symbols from the first Compilation should be considered null-oblivious. Currently that's handled by requiring that source Symbols are null-oblivious if the feature is disabled.

Reply:
Filed follow-up issue https://github.com/dotnet/roslyn/issues/26199 for general question of binding symbols differently when nullability feature is on/off.

Reply:
From our discussion, I'm leaving this this test and another one as skipped. 
When loading PE with an array and modifiers, we end up creating incorrect `TypeSymbolWithAnnotations`.

```C#
// this method is called by MetadataDecoder
        internal override TypeSymbol GetSZArrayTypeSymbol(PEModuleSymbol moduleSymbol, TypeSymbol elementType, ImmutableArray<ModifierInfo<TypeSymbol>> customModifiers)
        {
            if (elementType is UnsupportedMetadataTypeSymbol)
            {
                return elementType;
            }

            return ArrayTypeSymbol.CreateSZArray(moduleSymbol.ContainingAssembly, TypeSymbolWithAnnotations.Create(elementType, CSharpCustomModifier.Convert(customModifiers))); // the TypeSymbolWithAnnotations.Create should receive a module or a flag to indicate whether to treat unannotated reference types as non-null
        }
```

I'll leave a comment

---
In reply to: [181941297](https://github.com/dotnet/roslyn/pull/25928#discussion_r181941297) [](ancestors = 181941297)